src/main/scala/org/ergoplatform/nodeView/wallet/WalletCache.scala

```diff
@@ -67,20 +67,29 @@ object WalletCache {
     f
   }
 
-  def trackedBytes(trackedPubKeys: Seq[ExtendedPublicKey]): Seq[Array[Byte]] =
+  private def trackedBytes(trackedPubKeys: Seq[ExtendedPublicKey]): Seq[Array[Byte]] =
     trackedPubKeys.map(_.key.propBytes.toArray)
 
-  def trackedAddresses(trackedPubKeys: Seq[ExtendedPublicKey],
-                       addressEncoder: ErgoAddressEncoder): Seq[P2PKAddress] =
-    trackedPubKeys.map(pk => P2PKAddress(pk.key)(addressEncoder))
+  private def publicKeyAddresses(trackedPubKeys: Seq[ExtendedPublicKey],
+                                 addressEncoder: ErgoAddressEncoder): Seq[P2PKAddress] = {
+    // remove master key from visible addresses if following-up keys are from EIP-3 range
+    val pks = if (trackedPubKeys.size > 1
+      && trackedPubKeys.head.path.isMaster
+      && trackedPubKeys(1).path.isEip3) {
```

Comment:
Suggest to extract this condition into a named predicate.

Reply:
But why? As it is used once in a small method (these lines take like a half of the method), then readability will not be improved I guess.

Reply:
What this predicate means? Looks like some intermediate state of the WalletCache. If yes, why don't give it a name?

Reply:
@aslesarenko there's the comment (line 75) explaining what is checked in the code